framework/src/media/audio/audio_manager.c

```diff
@@ -487,9 +542,18 @@ static unsigned int resample_stream_out(audio_card_info_t *card, void *data, uns
 {
 	unsigned int used_frames = 0;
 	unsigned int resampled_frames = 0;
+	uint8_t device_channel_num = 0;
+	float rechanneling_ratio;
 	src_data_t srcData = { 0, };
 
-	srcData.channels_num = 2;
+	if (get_supported_channel_num(OUTPUT, &device_channel_num) != AUDIO_MANAGER_SUCCESS) {
```

Comment:
same as above, here we can get given channel number of device through the pcm_get_channels